src/collapse/mod.rs

```diff
@@ -185,12 +185,28 @@ mod tests_common {
     use std::fs::File;
     use std::io::{self, BufRead, Read};
     use std::path::{Path, PathBuf};
-    use std::time::Instant;
+    use std::time::{Instant, Duration};
 
     use libflate::gzip::Decoder;
 
     use super::*;
 
+    pub(crate) trait DurationExt {
+        fn as_nanos_compat(&self) -> u128;
+    }
+
+    impl DurationExt for Duration {
+        #[rustversion::since(1.33)]
+        fn as_nanos_compat(&self) -> u128 {
+            self.as_nanos()
+        }
+
+        #[rustversion::before(1.33)]
+        fn as_nanos_compat(&self) -> u128 {
```

Comment:
If we're writing this out anyway, I would keep just this one and get rid of the >= 1.33 impl. That also lets us avoid the `rustversion` dependency.